netbox/netbox/api/authentication.py

```diff
@@ -18,6 +22,17 @@ def authenticate_credentials(self, key):
         except model.DoesNotExist:
             raise exceptions.AuthenticationFailed("Invalid token")
 
+        # Update last used, but only once a minute. This reduces the write load on the db
+        if not token.last_used or (timezone.now() - token.last_used).total_seconds() > 60:
+            # If maintenance mode is enabled, assume the database is read-only, and disable updating the token's
+            # last_used time upon authentication.
+            if get_config().MAINTENANCE_MODE:
+                logger = logging.getLogger('netbox.auth.login')
+                logger.warning("Maintenance mode enabled: disabling update of token's last used timestamp")
+            else:
+                token.last_used = timezone.now()
+                token.save()
```

Comment:
It might be a bit faster to make a raw `update()` query to update the time.

```suggestion
                Token.objects.filter(pk=token.pk).update(last_used=timezone.now())
```